armi/reactor/reactors.py

```diff
@@ -2129,6 +2129,77 @@ def growToFullCore(self, cs):
 
         return converter
 
+    def checkGridPlateMaterialConsistency(self):
+        """check that the grid plates in the core are the same material"""
+        gridPlateBlocks = self.getBlocks(Flags.GRID_PLATE)
+        for ib, b in enumerate(gridPlateBlocks):
+            gridPlateComponent = b.getComponent(Flags.GRID_PLATE)
+            gridPlateMaterial = gridPlateComponent.material
+            if ib == 0:
+                refGridPlateMat = gridPlateMaterial
+                continue
+            if not isinstance(gridPlateMaterial, type(refGridPlateMat)):
+                runLog.warning(
+                    "Grid plate material for assembly is not consistent! Grid plate expansion/contraction assumes\n"
+                    f"a consistent material and will use {refGridPlateMat}. The inconsistency is:\n"
+                    f"Reference grid plate material = {refGridPlateMat}\n"
+                    f"Inconsistent grid plate material = {gridPlateMaterial} -- is in {b.parent}"
+                )
+
+    def calculateNewGridPlatePitchFromTemp(self, cs, isoTemp: float) -> float:
+        """calculate a grid plate pitch using a new bulk ave core inlet temperature in C
+
+        Parameters
+        ----------
+        cs
+            case settings
+        isoTemp : float
+            an isothermal temperature in C to apply to cs["Tin"]
+        """
+        self.checkGridPlateMaterialConsistency()
+        # retrieve the first grid plate (all the same material, so just get the first)
+        gridPlateBlock = self.getFirstBlock(blockType=Flags.GRID_PLATE)
+        # retrieve common grid plate component, material, temp, and pitch
+        gridPlateComponent = gridPlateBlock.getComponent(Flags.GRID_PLATE)
+        gridPlateMaterial = gridPlateComponent.material
+        currentGridPlatePitch = self.getAssemblyPitch()
+        currentGridPlateTemp = cs["Tin"]
```

Comment:
Is this true? Shouldn't this be the grid plate component temperature? Might it better if the initial grid plate temperature was just set on the blueprints as `Tinput` and then go from there?

Reply:
Discussed offline and addressed https://github.com/terrapower/armi/pull/993/commits/01ce852e7d7759ae463e6f3c612e031db1f77717